src/server/req_stat.c

```diff
@@ -578,18 +648,83 @@ req_stat_node(struct batch_request *preq)
 	preply->brp_choice = BATCH_REPLY_CHOICE_Status;
 	CLEAR_HEAD(preply->brp_un.brp_status);
 
-	if (type == 0) {		/*get status of the named node*/
-		rc = status_node(pnode, preq, &preply->brp_un.brp_status);
+	memset(&attr, 0, sizeof(attr));
 
-	} else {			/*get status of all nodes     */
+	if (preq->rq_extend != NULL) {
+		rc = sched_attr_def[SCHED_ATR_partition].at_decode(&attr, ATTR_partition, NULL, preq->rq_extend);
+		if (rc) {
+			(void) reply_free(preply);
+			req_reject(rc, bad, preq);
+			return;
+		}
+	}
 
+	if (type == 0) {		/*get status of the named node*/
+		if (preq->rq_extend != NULL) {
+			if (pnode->nd_attr[ND_ATR_partition].at_flags & ATR_VFLAG_SET) {
+				/*search for the partition in the attr*/
+				for (k = 0; k < attr.at_val.at_arst->as_usedptr; k++) {
+					if ((attr.at_val.at_arst->as_string[k] != NULL)
+							&& (!strcmp(
+									pnode->nd_attr[ND_ATR_partition].at_val.at_str,
+									attr.at_val.at_arst->as_string[k]))) {
+						consider_n = 1;
+						break;
+					}
+				}
+			}
+		} else {
+			if (preq->rq_conn == dflt_scheduler->scheduler_sock) {
+				if (!(pnode->nd_attr[ND_ATR_partition].at_flags & ATR_VFLAG_SET))
+					consider_n = 1;
+			} else {
+				consider_n = 1;
+			}
+		}
+		if (consider_n || preq->rq_conn != dflt_scheduler->scheduler_sock) {
+			rc = status_node(pnode, preq, &preply->brp_un.brp_status,
+					(preq->rq_extend ? &attr : NULL));
+			consider_n = 0;
+		}
+	} else {
+		/*get status of all nodes     */
 		for (i=0; i<svr_totnodes; i++) {
 			pnode = pbsndlist[i];
-
-			rc = status_node(pnode, preq,
-				&preply->brp_un.brp_status);
-			if (rc)
-				break;
+			if (preq->rq_extend != NULL) {
+				if (pnode->nd_attr[ND_ATR_partition].at_flags & ATR_VFLAG_SET) {
+					/*search for the partition in the attr*/
+					for (k = 0; k < attr.at_val.at_arst->as_usedptr; k++) {
+						if ((attr.at_val.at_arst->as_string[k] != NULL)
+								&& (!strcmp(
+										pnode->nd_attr[ND_ATR_partition].at_val.at_str,
+										attr.at_val.at_arst->as_string[k]))) {
+							consider_n = 1;
+							break;
+						}
+					}
+				}
+				if (!consider_n) {
+					continue;
+				}
+			} else {
+				if (preq->rq_conn == dflt_scheduler->scheduler_sock) {
+					if (!(pnode->nd_attr[ND_ATR_partition].at_flags & ATR_VFLAG_SET))
+						consider_n = 1;
+				} else {
+					consider_n = 1;
+				}
+			}
+			if (consider_n || preq->rq_conn != dflt_scheduler->scheduler_sock) {
```

Comment:
The above "if" condition is used for both of the following cases.

1. When scheduler is querying the queues
    In case of non default scheduler if its partition does not match with the queue's partition then 
    it won't hit the above if block. It exits the loop earlier i.e. it does not reach to this statement.
2. When user/admin list/print queues
     In case of non default scheduler we should actually list all the queues even if they are 
     associated with partition or not if user/admin does things like qmgr -c "print queue @default"

In order to satisfy above use cases I have done the changes accordingly.

Reply:
I thought we decided all the filtering was going to happen at the scheduler side.  Won't the first bullet not matter?